### ISSUE TITLE
elasticnet() stops the run when an inner fold leaves a response with 0 or 1 presences

### ISSUE THREAD

**Issue by gcol33** [gcol33/timesift#72]
## Summary

`elasticnet()` stops the whole run when an inner fold leaves a response with 0 or 1 presences. The
error comes out of `cv.glmnet` rather than from a timesift check, so it reads as an internal glmnet
failure rather than as a statement about the design that produced it.

## Observed

Benchmark cell `elasticnet-season-n300`, replicate 44, at `1e91a17`:

```
[elasticnet-season-n300] replicate 44: n=300, season, 12 candidates (274ec6304c8d), inner=5, outer=5, 0.1.1@1e91a17a
Error in lognet(x, is.sparse, y, weights, offset, alpha, nobs, nvars,  :
  one multinomial or binomial class has 1 or 0 observations; not allowed
Calls: bench_replicate ... FUN -> <Anonymous> -> cv.glmnet.raw -> glmnet -> lognet
Execution halted
```

## Repro

```
Rscript inst/benchmark/run.R --scale=full --cell=elasticnet-season-n300 --reps=44 \
  --out=/tmp/degenerate --pkg=.
```

It also fires on the smoke cell `smoke-elasticnet-none-n120` at replicate 1, where n=120 and a
prevalence of 0.10 make the degenerate fold much more likely, so that is the faster reproduction.

## Rate and impact

One task in 1600 at 300 units on the full-scale penalised block, and every smoke run of the
120-unit cell. The replicate is fixed by its design seed, so the same draw and the same degenerate
fold come back on a rerun: the replicate cannot be recovered by resubmitting, and the cell it
belongs to is permanently one replicate short. `elasticnet-season-n300` currently holds 199 of 200
for this reason.

## Expected

A degenerate inner fold is a property of the design, not an error in the fitter, so the package
should decide what it means rather than letting glmnet decide. Three behaviours seem defensible and
the choice is a design question rather than an obvious fix:

- skip the affected response for that fold and record that it was skipped, keeping the rest of the
  replicate;
- redraw the inner split so every response is fittable, and record that it was redrawn;
- stop, as now, but from a timesift check that names the response, the fold and the presence count,
  so the cause is legible without reading the glmnet call stack.

The third is the smallest change and would already have saved the diagnosis time here. The first is
the one that would not have cost a replicate.

## Notes

`.inner_folds()` already takes the response and the grouping, so it is the natural place to detect
the condition. Whether the same case can arise for `forest()` and the torch learners has not been
checked; only the penalised learner has been observed to hit it.

**Comment by gcol33** [comment]
Fixed in 35f1d83. I took a combination: redraw the inner split so every response is fittable wherever its count allows, and skip the response where it doesn't, recording that it was skipped.

- **Stratified inner folds.** They are now dealt for each response and stratified on it, through the same seeded deal `fold_map()` uses. That deal is extracted as `.seeded_deal()` / `_deal()`. Regenerating every fixture left the outer fold maps byte-identical. With five presences over five inner folds, each fold now holds exactly one; a plain deal put two in one fold in most draws.
- **Responses too rare to choose a penalty on.** A presence-absence response whose inner training sets can't each hold two of each outcome (the minimum glmnet's logistic path needs) is now predicted its share among the fitting units. That was already the rule for a response holding one outcome. The fit names such responses in `unfitted`, instead of stopping the run inside `cv.glmnet()`.
- **Python penalty criterion.** The Python side now chooses the penalty on the weighted held-out log loss, as `cv.glmnet()` chooses on the weighted deviance. It had been using accuracy, whose choice jumped by orders of magnitude with the fold draw: C = 0.046 against 1292 for the weighted and unweighted fits of the same response. The rare-response weighting test caught this under the new deal. With log loss both fits choose C = 21.5.

**Verified:** `elasticnet-season-n300` replicate 44 at full scale, the reproduction above, now completes (19.4 min). So does the smoke cell `smoke-elasticnet-none-n120` replicate 1.

**Other learners:** `forest()` and `stepwise()` do no inner cross-validation, and the torch learners fit the responses jointly with no inner split, so none of them has the failure path this issue describes.

The inner draw is not the one before, so an elastic-net fit's chosen penalty can differ from an earlier run's. NEWS says so.
